Boa.Constrictor/Screenplay/Caching/AnswerCache.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+
+namespace Boa.Constrictor.Screenplay
+{
+    /// <summary>
+    /// Caches answers to Questions.
+    /// </summary>
+    public class AnswerCache
+    {
+        #region Instance Variables
+
+        /// <summary>
+        /// Lock object for safe multi-threading.
+        /// </summary>
+        protected readonly object Lock = new object();
+
+        /// <summary>
+        /// The answer cache.
+        /// Maps Question objects to answer objects.
+        /// </summary>
+        protected IDictionary<IInteraction, object> Cache { get; set; }
```

Comment:
Any reason to not Initialize here?

`protected IDictionary<IInteraction, object> Cache { get; set; } = new Dictionary<IInteraction, object>();`

Reply:
I don't see why not.

Reply:
On second thought, I'd prefer to keep it the way it is. This style is more consistent with other classes in this project.

Reply:
The lock object is initialized in that way because it is `readonly`.